tests/utils/stl/test/config.py

```diff
@@ -9,24 +9,25 @@
 import os
 
 def configure(parameters, features, config, lit_config):
-  # Apply parameters to the configuration first, since parameters are things
-  # that we request explicitly and which might influence what features are
-  # implicitly made available next.
+  # Apply the actions supplied by parameters to the configuration first, since
+  # parameters are things that we request explicitly and which might influence
+  # what features are implicitly made available next.
   for param in parameters:
-    feature = param.getFeature(config, lit_config.params)
-    if feature:
-      feature.enableIn(config)
-      lit_config.note(
-        "Enabling Lit feature '{}' as a result of parameter '{}'".format(feature.getName(config), param.name))
+    actions = param.getActions(config, lit_config.params)
+    for action in actions:
+      action.applyTo(config)
+      #lit_config.note("Applied '{}' as a result of parameter '{}'".format(
+      #  action.pretty(config, lit_config.params),
+      #  param.pretty(config, lit_config.params)))
 
   # Then, apply the automatically-detected features.
-  printFeatures = []
   for feature in features:
-    if feature.isSupported(config):
-      feature.enableIn(config)
-      printFeatures.append(feature.getName(config))
-  printFeatures = ["'{}'".format(f) for f in sorted(printFeatures)]
-  lit_config.note("Enabling implicitly detected Lit features {}".format(', '.join(printFeatures)))
+    actions = feature.getActions(config)
+    for action in actions:
+      action.applyTo(config)
+      #lit_config.note("Applied '{}' as a result of implicitly detected feature '{}'".format(
+      #  action.pretty(config, lit_config.params),
+      #  feature.pretty(config)))

```

Comment:
Note that the above changes are copied from libcxx's `newconfig.py`. I commented out the `lit_config.note` bits because I don't think we want to see this notification spam for every run, but I'm leaving the code here to make it obvious to the next person that this was in fact copied from `newconfig.py`.